src/common/k8s-api/kube-object.store.ts

```diff
@@ -64,6 +64,24 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
   public readonly bufferSize: number = 50000;
   @observable private loadedNamespaces?: string[];
 
+  /**
+   * The field is used for configuring whether the edit button is displayed by default from
+   * {@link KubeObjectMenu} for detail views of instances of `T`.
+   *
+   * This should only be set once, in the constructor of base classes since the views will
+   * not be reactive to any changes to this value.
+   */
+  public readonly showEditButton = true;
```

Comment:
IMHO it's not good practice to mix UI logic to store/api level. Maybe these could be called as `isEditable` & `isRemovable` ?

Reply:
I agree that we should not mix them. But I don't see exactly how changing the name would help.

Maybe we could just extract this into some sort of extension API call which can be called to turn it off for any store. WDYT?

Reply:
`showEditButton` says that it affects the edit button visibility. `isEditable` is much more generic, edit button can use this information but so can other functionality. Small detail but I think we should use more generic names if possible.

Reply:
Okay will change.